feat(export-to-language) implement export to language for ruby RUBY-2873

Neilshweky opened the pull request:
This PR is for adding ruby as an output language. I left a bunch of comments on things I wasn't sure about.

addaleax reviewed (approved):
This looks good from the Compass side of things -- is there anything in the ruby templates/etc. itself that you think requires special attention from us? I don't know if any of us are really going to be helpful when it comes to reviewing ruby code or the driver API usage.

Neilshweky commented:
Hey @addaleax thanks so much for reviewing! I think some of the things you can help with are some of the comments I left which are more conceptual and have to do with other languages.

Neilshweky commented:
@addaleax I also just added Ruby to the export to language select in the compass UI, if you want to check that out (and rebased to get rid of the merge conflicts)

p-mongo commented on packages/bson-transpilers/test/yaml/bson-methods.yaml:
```diff
@@ -92,6 +100,12 @@ tests:
+                  '$ref' => 'coll',
+                  '$id' => BSON::ObjectId.new,
+                  '$db' => 'db'
+                }).database
```
The curly braces (`{}`) are redundant here, though this syntax does work.

p-mongo commented on packages/bson-transpilers/test/yaml/bson-methods.yaml:
```diff
@@ -543,6 +630,7 @@ tests:
           shell: ObjectId.fromDate(new Date('Thu, 03 Feb 1972 00:00:00 GMT'))
           # TODO: fix? this previously called ObjectId.createFromTime with a Date, but that function does not actually take a Date argument
           # object: new ObjectId('03ede8800000000000000000')
+          ruby: BSON::ObjectId.from_time(Time.new(1972, 2, 3, 0, 0, 0).utc)
```
There's a `Time.utc` method if you need the UTC time.

p-mongo reviewed (approved):
:+1:  The Ruby output looks good to me.
